lib/BootBot.js

```diff
@@ -36,27 +36,46 @@ class BootBot extends EventEmitter {
     this.app.use(this.webhook, bodyParser.json({ verify: this._verifyRequestSignature.bind(this) }));
     this._hearMap = [];
     this._conversations = [];
+    this.httpsServer = undefined;
   }
 
   /**
-   * Starts the express server on the specified port. Defaults port to 3000.
+   * Starts the express or HTTPS server on the specified port. Defaults port to 3000.
    * @param {Number} [port=3000]
+   * @param {{ key: string, cert: string }} [credentials=undefined] The credentials needed to run the https server
    */
-  start(port) {
+  start(port, credentials) {
     this._initWebhook();
     this.app.set('port', port || 3000);
-    this.server = this.app.listen(this.app.get('port'), () => {
-      const portNum = this.app.get('port');
-      console.log('BootBot running on port', portNum);
-      console.log(`Facebook Webhook running on localhost:${portNum}${this.webhook}`);
-    });
+    if (credentials) {
+      // Start the HTTPS server
+      this.httpsServer = https.createServer(credentials, this.app);
```

Comment:
What are the benefits of using a separate variable here (`httpsServer`) instead of assigning it to `this.server`? That way you wouldn't need the if statement in the `close()` method.